infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/PodTerminationGracePeriodProvisioner.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+
+public class PodTerminationGracePeriodProvisioner implements ConfigurationProvisioner {
+  private final long graceTerminationPeriodSec;
+
+  @Inject
+  public PodTerminationGracePeriodProvisioner(
+      @Named("che.infra.kubernetes.pod.termination_grace_period_sec")
+          long graceTerminationPeriodSec) {
+    this.graceTerminationPeriodSec = graceTerminationPeriodSec;
+  }
+
+  @Override
+  public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
+      throws InfrastructureException {
+    for (Pod pod : k8sEnv.getPods().values()) {
+      if (!isTerminationGracePeriodSet(pod)) {
```

Comment:
Maybe it would be better to override user-defined grace period. And do not override only if the grace period is configured to `0`.
If not then it makes sense to document it and maybe rename configuration property like `default_termination_grace_period_sec`.

I vote for overriding user-defined grace period.
@garagatyi @ibuziuk

Reply:
Not sure we should override the value specified by a user. What for? I opt for overriding only values that are not specified. About renaming I think current name is OK, and name with prefix default would be OK too.
In case we need to set it to 0, in any case, we will be able to add it as a separate flag later.

Reply:
> What for?

To be able to guarantee that workspace will be stopped(or start will be interrupted) in predictable time.

Reply:
And what if user's app needs more time, e.g. 1 minute?

Reply:
The situation is similar to the following: a user's container needs more time to start than workspace start timeout is configured.
Since removing of pods is the last phase of workspace stop, and we remove all pods asynchronously, I'm OK not to override a value defined by a user. But let's document it in che.properties.

Reply:
+1 to document it. Let's start without rewriting user's timeouts and then we will figure out whether we need to change it